src/Tests/Microsoft.NET.Build.Tests/WorkloadTests.cs

```diff
@@ -237,5 +237,67 @@ public void It_should_import_aliased_pack()
                 .Should()
                 .BeEquivalentTo("true");
         }
+
+        [Fact]
+        public void It_should_get_suggested_workload_by_FindAllReferenceWorkload_target()
+        {
+            var mainProject = new TestProject()
+            {
+                Name = "MainProject", TargetFrameworks = "net6.0-android", IsSdkProject = true, IsExe = true
+            };
+
+            var testAsset = _testAssetsManager
+                .CreateTestProject(mainProject);
+
+            var getValuesCommand =
+                new GetValuesCommand(testAsset, "_ResolvedSuggestedWorkloads", GetValuesCommand.ValueType.Item);
+            getValuesCommand.DependsOnTargets = "_FindAllReferenceWorkload";
+            getValuesCommand.ShouldRestore = false;
+
+            getValuesCommand.Execute("/p:SkipResolvePackageAssets=true")
+                .Should()
+                .Pass();
+
+            getValuesCommand.GetValues()
+                .Should()
+                .BeEquivalentTo("microsoft-android-sdk-full", "microsoft-net-runtime-mono-tooling");
+        }
+
+        [Fact]
+        public void Given_multi_target_It_should_get_suggested_workload_by_FindAllReferenceWorkload_target()
```

Comment:
This tests a multi-targeted project referencing a multi-targeted project, but they both require the same workloads.  I think we should cover more cases, such as:

- `net6.0` referencing `net6.0;net6.0-android;net6.0-ios`.  With the way we've implemented workload restore, this will require the ios and android workloads, even though they are not strictly required to build the first project.
- `net6.0;net6.0-ios` referencing `net6.0;net6.0-android`.  This is similar but has different workloads returned by each project
- `net6.0` referencing `net6.0`, where the referenced project has `RunAOTCompilation` set to true.  This validates when a referenced project requires a workload and there's no multi-targeting.